app/controllers/user/emails_controller.rb

```diff
@@ -0,0 +1,22 @@
+class User::EmailsController < ApplicationController
+  load_and_authorize_resource :email, class: User::Email.name
+
+  # Set an email as the primary email
+  def set_primary
+    existing_primary_email = current_user.emails.find(&:primary?)
```

Comment:
Not ACID. Wrap in a transaction.

Also, it is a bad idea to have multiple resources accessed in an action.